logistics/scripts/modules/settings/settings-management.psm1

```diff
@@ -0,0 +1,351 @@
+                            Refresh = "true"
+                        }
+                    }
+                }
+            }
+        }
+        "EdFi.Ods.SwaggerUI"        = @{
+            SwaggerUIOptions = @{
+                OAuthConfigObject = @{
+                    ClientId     = $populatedKey
+                    ClientSecret = $populatedSecret
+                }
+            }
+        }
+    }
+}
+
+function Get-ConnectionStringKeyByDatabaseTypes {
+    return @{
+        Ods      = 'EdFi_Ods'
+        Admin    = 'EdFi_Admin'
+        Security = 'EdFi_Security'
+        Master   = 'EdFi_Master'
+    }
+}
+
+function Get-ConnectionStringsByEngine {
+    return  @{
+        SQLServer  = @{
+            ConnectionStrings = @{
+                ((Get-ConnectionStringKeyByDatabaseTypes)['Ods'])      = "Server=(local); Trusted_Connection=True; Database=EdFi_{0};"
+                ((Get-ConnectionStringKeyByDatabaseTypes)['Admin'])    = "Server=(local); Trusted_Connection=True; Database=EdFi_Admin;"
+                ((Get-ConnectionStringKeyByDatabaseTypes)['Security']) = "Server=(local); Trusted_Connection=True; Database=EdFi_Security; Persist Security Info=True;"
+                ((Get-ConnectionStringKeyByDatabaseTypes)['Master'])   = "Server=(local); Trusted_Connection=True; Database=master;"
+            }
+            DatabaseTemplate  = @{
+                MinimalTemplateScript   = 'EdFiMinimalTemplate'
+                PopulatedTemplateScript = 'GrandBend'
+            }
+        }
+        PostgreSQL = @{
+            ConnectionStrings = @{
+                ((Get-ConnectionStringKeyByDatabaseTypes)['Ods'])      = "Host=localhost; Port=5432; Username=postgres; Database=EdFi_{0};"
+                ((Get-ConnectionStringKeyByDatabaseTypes)['Admin'])    = "Host=localhost; Port=5432; Username=postgres; Database=EdFi_Admin;"
+                ((Get-ConnectionStringKeyByDatabaseTypes)['Security']) = "Host=localhost; Port=5432; Username=postgres; Database=EdFi_Security;"
+            }
+            DatabaseTemplate  = @{
+                MinimalTemplateScript   = 'PostgreSQLMinimalTemplate'
+                PopulatedTemplateScript = 'PostgreSQLPopulatedTemplate'
+            }
+        }
+    }
+}
+
+function Get-SubtypesByFeature {
+    return @{
+        changeQueries               = 'changes'
+        ownershipBasedAuthorization = 'RecordOwnership'
+    }
+}
+
+function Test-AppSettings {
+    param(
+        [Parameter(ValueFromPipeline = $true)]
+        [string] $Path
+    )
+
+    if (-not (Test-Path $Path)) { return $false }
+
+    try {
+        Get-Content $Path | ConvertFrom-Json | Out-Null
+        return $true
+    }
+    catch {
+        return $false
+    }
+}
+
+function Assert-ValidAppSettings([string[]] $SettingsFiles = (Get-ChildItem "$(Get-RepositoryRoot 'Ed-Fi-ODS-Implementation')\**\appsettings*.json" -Recurse)) {
+    $result = @()
+
+    foreach ($file in $SettingsFiles) {
+        if (Test-AppSettings $file) {
```

Comment:
Does it make sense to change this function to return the same object this larger method is returning?  Wrap the file name, success, and the exception if any?  This would let you avoid parsing the file into json twice in order to get the exception information, and likely makes the single file function more useful outside of this usage.

Reply:
modified this and got rid of the other function, it's now only parsing once